File: src/main/java/com/ggardet/patient/internal/entity/Patient.java
package com.ggardet.patient.internal.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDate;
import java.util.UUID;

@Entity
@Data
@Table(name = "patient", schema = "patients")
public class Patient {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    
    @Column(nullable = false)
    private String firstname;
    
    @Column(nullable = false)
    private String lastname;
    
    @Column(name = "birth_date", nullable = false)
    private LocalDate birthDate;
    
    @Column(name = "social_security_number")
    private String socialSecurityNumber;
    
    @Column(name = "phone_number")
    private String phoneNumber;
    
    @Column
    private String email;
    
    @Column(name = "hospital_id")
    private UUID hospitalId;
    
    @Column(name = "service_id") 
    private UUID serviceId;
}

File: src/main/kotlin/com/ggardet/patient/internal/entity/Patient.kt
package com.ggardet.patient.internal.entity

import jakarta.persistence.*
import java.time.LocalDate
import java.util.*

@Entity
@Table(name = "patient", schema = "patients")
data class Patient(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    var id: UUID? = null,
    
    @Column(nullable = false)
    var firstname: String,
    
    @Column(nullable = false)
    var lastname: String,
    
    @Column(name = "birth_date", nullable = false)
    var birthDate: LocalDate,
    
    @Column(name = "social_security_number")
    var socialSecurityNumber: String? = null,
    
    @Column(name = "phone_number")
    var phoneNumber: String? = null,
    
    @Column
    var email: String? = null,
    
    @Column(name = "hospital_id")
    var hospitalId: UUID? = null,
    
    @Column(name = "service_id") 
    var serviceId: UUID? = null
) {
    // JPA requires a no-arg constructor
    constructor() : this(
        firstname = "",
        lastname = "",
        birthDate = LocalDate.now()
    )
}